debian/rules

```diff
@@ -24,17 +24,67 @@ export DH_GOLANG_INSTALL_ALL := 1
 # Tests needing sudo will be skipped automatically
 export ADSYS_SKIP_INTEGRATION_TESTS=1
 
+# Do not run Rust tests from Golang, as they're run independently via
+# dh-cargo, allowing integration with vendored deps, etc...
+export ADSYS_SKIP_RUST_TESTS_IN_GOLANG=1
+
+# The following definitions are necessary because of the manual steps
+# we need to do to work around some issues with either dh-cargo,
+# the wrapper, or cargo
+include /usr/share/rustc/architecture.mk
+CARGO := /usr/share/cargo/bin/cargo
+ADSYS_MOUNT := internal/policies/mount/adsys_mount
+
+# Needed for Rust vendored sources tracking
+DH_CARGO_VENDORED_SOURCES := /usr/share/cargo/bin/dh-cargo-vendored-sources
+export CARGO_VENDOR_DIR = vendor_rust
+
 %:
 	dh $@ --buildsystem=golang --with=golang,apport
 
 override_dh_auto_clean:
 	dh_auto_clean
+	dh_auto_clean --buildsystem=cargo
 	# Create the vendor directory when building the source package
 	[ -d vendor/ ] || go mod vendor
+	# system-deps crate includes an empty .a file which is filtered out when
+	# building the source package and cargo gets confused when trying to verify
+	# integrity. So we just put in an empty checksum file instead.
+	if [ ! -d $(CARGO_VENDOR_DIR)/ ]; then \
+		$(CARGO) vendor $(CARGO_VENDOR_DIR); \
+		cp debian/cargo-checksum.json $(CARGO_VENDOR_DIR)/system-deps/.cargo-checksum.json; \
+		[ ! -e $(DH_CARGO_VENDORED_SOURCES) ] || $(DH_CARGO_VENDORED_SOURCES); \
+	fi
+	# That last line above is to warn whoever's doing the packaging that they
+	# need to update the Vendored-Sources field. The initial packaging work
+	# didn't include said field since the vendoring is done on the fly.
+
+override_dh_auto_configure:
+	dh_auto_configure
+	# Normally we keep the lockfile around when using vendored deps, except we
+	# had to mess with the checksums of system-deps and so cargo complains.
+	# That doesn't actually matter since dh-cargo configured cargo not to query
+	# the network for deps, instead relying on the vendored ones.
+	rm -f Cargo.lock
```

Comment:
I’m not a big fan of this. It means that running debuild in the source directory removes some files in the source package itself, which is packaging nono in general. Also, we can’t `dpkg-buildpackage -nc` to continue a failed build after some fixes we wanted too if we removed the vendor directory.

Of course `Cargo.lock` is less useful from a security team perspective as `XS-Vendored-Sources-Rust` is keeping the exact version of vendored dependency for us, but I think this issue may be considered.

Of course,